### PR TITLE
Cover cmds/core/dd

### PR DESCRIPTION
**Pull request by walterchris**

### CODE REVIEW

File: pkg/progress/progress.go [hunk 1]
```diff
@@ -0,0 +1,89 @@
+// Copyright 2021 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
```
Comment (jensdrenhaus) [inline pkg/progress/progress.go:4]: maybe short description what this package is about

File: pkg/progress/progress.go [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+type progressData struct {
+	mode         string // one of: none, xfer, progress
+	start        time.Time
+	end          time.Time
+	endTimeMutex sync.Mutex
+	variable     *int64 // must be aligned for atomic operations
+	quit         chan struct{}
+}
+
+// Begin - start a progress routine
```
Comment (jensdrenhaus) [inline pkg/progress/progress.go:25]: can you explain the input parameters in

File: pkg/progress/progress.go [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+					p.endTimeMutex.Unlock()
+					return
+				}
+			}
+		}()
+	}
+	return p
+}
+
+func (p *progressData) End() {
```
Comment (jensdrenhaus) [inline pkg/progress/progress.go:56]: comment with description of public function

File: pkg/progress/progress_test.go [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			someVariable := int64(1)
+			p := Begin(tt.mode, &someVariable)
+
+			if p == nil {
+				t.Errorf("%s failed - struct is nil", tt.name)
+			}
```
Comment (jensdrenhaus) [inline pkg/progress/progress_test.go:40]: Take a look at [https://pkg.go.dev/github.com/stretchr/testify/require](https://pkg.go.dev/github.com/stretchr/testify/require)
But that's just a question of style / preference

**Review by depate — DISMISSED** [review]
looks good to me when Jens' comments are resolved.

**Pull request closed without merging**